### PR TITLE
Add Global sidebar header

### PR DESCRIPTION
**Pull request by eoigal**
This PR updates the global sidebar to include a header that matches the latest designs.

### New styles 
Screenshot 2024-02-16 at 16 33 30

### Tooltip styling
Screenshot 2024-02-16 at 16 33 43

This is a follow up;
* https://github.com/Automattic/wp-calypso/pull/87514
* https://github.com/Automattic/wp-calypso/pull/87515

### Things of note
* This adds new classes to the layout div if the global sidebar is visible
* This does not update the global sidebar body to have the selected or hover styles, this will be done in follow up PR
* The search functionality is not working as it is in https://github.com/Automattic/wp-calypso/pull/86766 because there is some work being done on command palette - for now this is a placeholder until those changes have been done.
* This does not include the global sidebar footer

### CODE REVIEW

File: client/layout/global-sidebar/menu-items/menu-item.jsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+				preloadedRef.current = true;
+				preloadSection();
+			}
+		};
+
+		const renderChildren = () => {
+			return (
+				<Fragment>
+					{ hasUnseen && (
+						<span className="sidebar__item-bubble" aria-label="You have unseen content" />
```
Comment (ebinnion) [inline client/layout/global-sidebar/menu-items/menu-item.jsx:36]: Does this actually display anything? What would be unseen about domains or sites?

File: client/layout/global-sidebar/menu-items/search/icon.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+export const SearchIcon = () => (
```
Comment (ebinnion) [inline client/layout/global-sidebar/menu-items/search/icon.jsx:3]: I think that we should use the search icon from core for this instead: https://wordpress.github.io/gutenberg/?path=/docs/icons-icon--docs

**Review by ebinnion — APPROVED** [review]
I left some minor feedback. We should fix the search icon, but it seems reasonable if that happens in a follow-up PR, so going to approve.

Tested functionally and this works well with and without flag. No critical comments on code.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11359927

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @eoigal for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11359927).

**Pull request merged**